src/main/java/it/gov/pagopa/admissibility/service/commands/operations/DeleteInitiativeServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package it.gov.pagopa.admissibility.service.commands.operations;
+
+import it.gov.pagopa.admissibility.connector.repository.DroolsRuleRepository;
+import it.gov.pagopa.admissibility.connector.repository.InitiativeCountersRepository;
+import it.gov.pagopa.admissibility.connector.repository.OnboardingFamiliesRepository;
+import it.gov.pagopa.admissibility.utils.AuditUtilities;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+@Service
+@Slf4j
+public class DeleteInitiativeServiceImpl implements DeleteInitiativeService{
+
+    private final DroolsRuleRepository droolsRuleRepository;
+    private final InitiativeCountersRepository initiativeCountersRepository;
+    private final OnboardingFamiliesRepository onboardingFamiliesRepository;
+    private final AuditUtilities auditUtilities;
+
+    public DeleteInitiativeServiceImpl(DroolsRuleRepository droolsRuleRepository, InitiativeCountersRepository initiativeCountersRepository, OnboardingFamiliesRepository onboardingFamiliesRepository, AuditUtilities auditUtilities) {
+        this.droolsRuleRepository = droolsRuleRepository;
+        this.initiativeCountersRepository = initiativeCountersRepository;
+        this.onboardingFamiliesRepository = onboardingFamiliesRepository;
+        this.auditUtilities = auditUtilities;
+    }
+
+    @Override
+    public Mono<String> execute(String initiativeId) {
+        log.info("[DELETE_INITIATIVE] Starting handle delete initiative {}", initiativeId);
+        return  deleteDroolsRule(initiativeId)
+                .then(deleteInitiativeCounters(initiativeId))
+                .then(deleteOnboardingFamilies(initiativeId))
+                .then(Mono.just(initiativeId));
+    }
+
+    private Mono<Void> deleteDroolsRule(String initiativeId) {
+        return droolsRuleRepository.deleteById(initiativeId)
+                .doOnSuccess(d -> {
```

Comment:
after deleting drools rule, we should refresh Drools container calling (using flatMap as operator)
it.gov.pagopa.admissibility.service.onboarding.OnboardingContextHolderServiceImpl#refreshKieContainerCacheMiss
(make it public and add it to its interface)